test/unit/test_sdb.c

```diff
@@ -219,6 +374,10 @@ int all_tests() {
 	mu_run_test (test_sdb_list_big);
 	mu_run_test (test_sdb_foreach_filter);
 	mu_run_test (test_sdb_copy);
+	mu_run_test (test_sdb_text_save_simple);
+	mu_run_test (test_sdb_text_load_simple);
+	mu_run_test (test_sdb_text_save);
+	mu_run_test (test_sdb_text_load);
```

Comment:
could you also try running some tests with invalid projects to load? As this will be used for projects, I think we should be careful to ensure the loading of a SDB text is as safe as possible.

Reply:
There actually aren't too many cases of invalid sdb text. There are some things that are nonsense, but the parser just handles like `\r` newlines, multiple `/` or escaping chars that don't have to be escape in the respective scope.
The only real "invalid" thing is a line that does not start with `/` and does not contain a non-escaped `=` which would be just ignored.
I've added some tests for these things.

Reply:
What about things like `\` and then a newline? I think the newline would become part of the token, right? Is that ok?

Reply:
Good idea. I think it will ignore the `\` and treat the newline as record separator, but I have to check.

Reply:
Yep, added a test and it just ignores it.